receivers/discord/discord.go

```diff
@@ -202,52 +201,62 @@ func (d Notifier) SendResolved() bool {
 	return !d.GetDisableResolveMessage()
 }
 
-func (d Notifier) constructAttachments(ctx context.Context, as []*types.Alert, embedQuota int) []discordAttachment {
-	attachments := make([]discordAttachment, 0)
+func (d Notifier) constructAttachments(ctx context.Context, alerts []*types.Alert, embedQuota int) []discordAttachment {
+	attachments := make([]discordAttachment, 0, embedQuota)
+	for _, alert := range alerts {
+		// Check if the image limit has been reached at the start of each iteration.
+		if embedQuota < 1 {
+			break
+		}
 
-	_ = images.WithStoredImages(ctx, d.log, d.images,
-		func(index int, image images.Image) error {
-			if embedQuota < 1 {
-				return images.ErrImagesDone
+		attachment, err := d.getAttachmentFromURL(ctx, alert)
+		if err != nil {
+			if !errors.Is(err, images.ErrImagesNoURL) {
+				// If there's an error, continue iterating through the other alerts.
+				continue
```

Comment:
shouldn't we log error in this case? At least with warn severity

Reply:
I would go with inverse approach

Reply:
What do you mean by "inverse approach"? I agree with logging some warning if there's an error.

Reply:
```
if errors.Is(err, images.ErrImagesNoURL) {
     // The image has no public URL, use the bytes for the attachment.
    attachment, err = d.getAttachmentFromBytes(ctx, alert)    
}
if err != nil {
   d.log.Warn(...., err)
}